packages/legend-query/src/components/QueryBuilderResultPanel.tsx

```diff
@@ -39,17 +42,244 @@ import {
   ActionAlertType,
   EDITOR_LANGUAGE,
   ExecutionPlanViewer,
+  instanceValue_changeValue,
+  instanceValue_changeValues,
   PARAMETER_SUBMIT_ACTION,
   TAB_SIZE,
   TextInputEditor,
   useApplicationStore,
 } from '@finos/legend-application';
-import { isBoolean } from '@finos/legend-shared';
+import { assertErrorThrown, isBoolean } from '@finos/legend-shared';
+import { forwardRef, useState } from 'react';
+import type { CellMouseOverEvent } from '@ag-grid-community/core';
+import {
+  QueryBuilderDerivationProjectionColumnState,
+  QueryBuilderProjectionColumnState,
+} from '../stores/QueryBuilderProjectionState.js';
+import {
+  type QueryBuilderPostFilterTreeNodeData,
+  PostFilterConditionState,
+  QueryBuilderPostFilterTreeConditionNodeData,
+} from '../stores/QueryBuilderPostFilterState.js';
+import {
+  QueryBuilderPostFilterOperator_Equal,
+  QueryBuilderPostFilterOperator_NotEqual,
+} from '../stores/postFilterOperators/QueryBuilderPostFilterOperator_Equal.js';
+import {
+  QueryBuilderPostFilterOperator_In,
+  QueryBuilderPostFilterOperator_NotIn,
+} from '../stores/postFilterOperators/QueryBuilderPostFilterOperator_In.js';
+import type { QueryBuilderPostFilterOperator } from '../stores/QueryBuilderPostFilterOperator.js';
+
+const QueryBuilderResultContextMenu = observer(
+  forwardRef<
+    HTMLDivElement,
+    {
+      event: CellMouseOverEvent | null;
+      queryBuilderState: QueryBuilderState;
+    }
+  >(function QueryBuilderResultContextMenu(props, ref) {
+    const { event, queryBuilderState } = props;
+    const applicationStore = useApplicationStore();
+    const postFilterEqualOperator = new QueryBuilderPostFilterOperator_Equal();
+    const postFilterInOperator = new QueryBuilderPostFilterOperator_In();
+    const postFilterNotEqualOperator =
+      new QueryBuilderPostFilterOperator_NotEqual();
+    const postFilterNotInOperator = new QueryBuilderPostFilterOperator_NotIn();
+    const postFilterState = queryBuilderState.postFilterState;
+    const theProjectionColumnState =
+      queryBuilderState.fetchStructureState.projectionState.columns
+        .filter((c) => c.columnName === event?.column.getColId())
+        .concat(
+          queryBuilderState.fetchStructureState.projectionState.aggregationState.columns
+            .filter((c) => c.columnName === event?.column.getColId())
+            .map((ag) => ag.projectionColumnState),
+        )[0];
+
+    const getExistingPostFilterNode = (
```

Comment:
Your method return nodes not a singular node. We can either rename the method or return a singular node